src/taskmanager.cpp

```diff
@@ -430,11 +435,42 @@ void TaskManager::findMatchingArchives(QString jobPrefix)
     emit matchingArchives(matching);
 }
 
+bool TaskManager::waitForOnline()
+{
+    for(int i = 0; i < NETWORK_UP_ATTEMPTS; i++)
+    {
+        QTcpSocket sock;
+        sock.connectToHost(NETWORK_HOST, NETWORK_PORT);
+        // Multiple calls to waitForConnected() do not accumulate time, so we
+        // use ::sleep() below.
+        if(sock.waitForConnected(500))
```

Comment:
You do instantiate a new QTcpSocket in the loop here so I think the waitForConnect() will actually wait in that case. How about try and sleep for the NETWORK_UP_SLEEP here and see how it goes? Instead of accumulating more wait time than intended.

Reply:
Unfortunately it doesn't work even with the new `QTcpSocket`s.  :(

```
unknown(0): top of NETWORK_UP_ATTEMPTS "10:56:34"
unknown(0): calling sock.waitForConnected( 60000 )
unknown(0): top of NETWORK_UP_ATTEMPTS "10:56:39"
unknown(0): calling sock.waitForConnected( 60000 )
unknown(0): top of NETWORK_UP_ATTEMPTS "10:56:39"
unknown(0): calling sock.waitForConnected( 60000 )
```

also note that it only waits 5 seconds the first time, not 60!  Based on that flakiness, I figured that `QThread::sleep()` was much more reliable.

As for the wait time... 500 milliseconds is not going to change much, so I figured it wasn't worth worrying about.  :)